Library/Homebrew/diagnostic.rb

```diff
@@ -1111,6 +1112,20 @@ def check_tmpdir
         "TMPDIR #{tmpdir.inspect} doesn't exist." unless tmpdir.nil? || File.directory?(tmpdir)
       end
 
+      def check_tmpdir_executable
+        Tempfile.open("homebrew_check_tmpdir_executable", HOMEBREW_TEMP) do |f|
+          f.write "#!/bin/sh\necho hi\n"
+          f.chmod 0700
+          unless system f.path
+            <<-EOS.undent
+              The directory #{HOMEBREW_TEMP} does not permit executing programs.
+              It is likely mounted "noexec". Please add the following to your ~/.bashrc:
```

Comment:
Instead of a hardcoded `"~/.bashrc"`, there's code somewhere to guess which shell the user is running, and return the appropriate profile path (in case it's not bash). I think it's `"#{shell_profile}"`.